contracts/validator/contract/XDCValidatorV2.sol

```diff
@@ -0,0 +1,521 @@
+                break;
+            }
+        }
+
+        // Optionally, consider adjusting ownerCount if needed
+        if (ownedCandidates.length == 0) {
+            // If specific logic is needed to manage the owners array, implement here
+            deleteOwner(msg.sender);
+        }
+
+        uint256 cap = validatorsState[_candidate].voters[msg.sender];
+        validatorsState[_candidate].cap = validatorsState[_candidate].cap.sub(
+            cap
+        );
+        validatorsState[_candidate].voters[msg.sender] = 0;
+        // refunding after resigning X blocks
+        uint256 withdrawBlockNumber = candidateWithdrawDelay.add(block.number);
+        withdrawsState[msg.sender].caps[withdrawBlockNumber] = withdrawsState[
+            msg.sender
+        ].caps[withdrawBlockNumber].add(cap);
+        withdrawsState[msg.sender].blockNumbers.push(withdrawBlockNumber);
+        emit Resign(msg.sender, _candidate);
+    }
+
+    // voteInvalidKYC : any candidate can vote for invalid KYC i.e. a particular candidate's owner has uploaded a bad KYC.
+    // On securing 75% votes against an owner ( not candidate ), owner & all its candidates will lose their funds.
+    function voteInvalidKYC(
+        address _owner
+    ) public onlyValidCandidate(msg.sender) {
+        address candidateOwner = getCandidateOwner(msg.sender);
+
+        require(!hasVotedInvalid[candidateOwner][_owner], "Already voted");
+        hasVotedInvalid[candidateOwner][_owner] = true;
+        invalidKYCCount[_owner] += 1;
+        if ((invalidKYCCount[_owner] * 100) / getOwnerCount() >= 75) {
+            // 75% owners say that the KYC is invalid
+            address[] memory allMasternodes = new address[](
+                candidates.length - 1
+            );
+            uint count = 0;
+            uint j = 0;
+            address[] memory newCandidates = new address[](candidates.length);
+            invalidOwner[_owner] = true;
+            for (uint i = 0; i < candidates.length; i++) {
+                if (getCandidateOwner(candidates[i]) == _owner) {
+                    // logic to remove cap.
+                    candidateCount = candidateCount.sub(1);
+                    allMasternodes[count++] = candidates[i];
+
+                    invalidCandidate[candidates[i]] = true;
+                    //delete voters[candidates[i]]; in blacklist no need use in the future
+                    // for (uint256 y = 0; y < voters[candidates[i]].length; y++) {
+                    //     //delete mapping
+                    //     validatorsState[candidates[i]].voters[
+                    //         voters[candidates[i]][y]
+                    //     ] = 0;
+                    //     voters[candidates[i]][y] = address(0);
+                    // }
+                    // voters[candidates[i]] = removeZeroAddresses(
+                    //     voters[candidates[i]]
+                    // );
+
+                    delete validatorsState[candidates[i]];
```

Comment:
`delete validatorsState[candidates[i]];` will drop a element `candidates[i]` in map `validatorsState`. But `validatorsState[candidates[i]]` points to the struct `ValidatorState`. And `ValidatorState` contains a map `voters`. So  delete validatorsState[candidates[i]]` will not delete `validatorsState[candidates[i]].voters`.

Reply:
right , but There is something else here ， i already dm you